job/dhcp.go

```diff
@@ -83,6 +89,10 @@ func (j Job) isPXEAllowed() bool {
 	return j.instance.AllowPXE
 }
 
+func (j Job) areWeProvisioner() bool {
+	return j.hardware.HardwareProvisioner() == j.ProvisioningEngineName()
```

Comment:
Would it make sense to return true if its not set? This way we don't *require* this setting?